pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -310,6 +311,63 @@ func newOpenStack(cfg Config) (*OpenStack, error) {
 	return &os, nil
 }
 
+func configFromEnv() (cfg Config, ok bool) {
+	cfg.Global.AuthUrl = os.Getenv("OS_AUTH_URL")
+
+	cfg.Global.TenantId = os.Getenv("OS_TENANT_ID")
+	// Rax/nova _insists_ that we don't specify both tenant ID and name
+	if cfg.Global.TenantId == "" {
+		cfg.Global.TenantName = os.Getenv("OS_TENANT_NAME")
+	}
+
+	cfg.Global.Username = os.Getenv("OS_USERNAME")
+	cfg.Global.Password = os.Getenv("OS_PASSWORD")
+	cfg.Global.Region = os.Getenv("OS_REGION_NAME")
+
+	cfg.Global.TenantName = os.Getenv("OS_TENANT_NAME")
+	if cfg.Global.TenantName == "" {
+		cfg.Global.TenantName = os.Getenv("OS_PROJECT_NAME")
```

Comment:
may have already retrieved this value above.
What do you think about `os.Getenv` **all** of the expected env vars at once in the beginning of this func, then after apply semantics regarding which are required, which can be defaulted, etc?

Reply:
I think it sounds good to me and I will effect. @dims I know this change is not related to the e2e tests, I hope its okay to commit this change with these e2e tests.